### DISCUSSION TITLE
How to make window_background_opacity function similar to kitty's

### DISCUSSION THREAD

**Discussion by hthuong09** [wezterm/wezterm#3280] [Q&A] [answered]
Hi,

I'm migrating from kitty to wezterm and trying to figure out how to make some options on kitty work
In kitty, I have this configuration
```
background_opacity 0.989
```

In wezterm, I have this
```
window_background_opacity = 0.989
```

It works perfectly but when I open vim, the opacity is not applied anymore.

In kitty, I think if vim's background matches the terminal background, it will automatically apply opacity.
Is there any way to achieve this in wezterm?

**Comment by nekowinston** [comment]
```lua
{
  window_background_opacity = 0.989,
  text_background_opacity = 0.989,
}
```
But this will cause rendering issues, so it's best to set your vim to hightlight `guibg=NONE`, e.g.:
```
hi! Normal ctermbg=NONE guibg=NONE
```

**Comment by hthuong09** [comment, marked as answer]
Thanks for the quick reply @nekowinston.

For the `text_background_opacity`. I'm not sure if I'm understanding it correctly. But this applies to all backgrounds of all cells, not just the one that has the same color as the terminal background.
> The text_background_opacity setting specifies the alpha channel value to use for the background color of cells other than the default background color.

This text opacity is also a layer above the background so setting it to `0.989` will not have the same as transparency level as background opacity. I need to set it to 0 to make it have the same transparency as the background but again the other cells' background color went wrong.

> But this will cause rendering issues, so it's best to set your vim to hightlight guibg=NONE, e.g.:

Yup, I know this. If there are no other ways then I will use this as it breaks a feel highlight in my color scheme and needs some customization.
